packages/gestalt/src/Avatar.js

```diff
@@ -84,6 +84,12 @@ export default class Avatar extends React.PureComponent<AvatarProps, State> {
 
   handleImageError = () => this.setState({ isImageLoaded: false });
 
+  componentDidUpdate = (prevProps: *) => {
```

Comment:
This is better but it will cause an extra render because React will handle it as follows:
New src comes in -> React renders Avatar (1) -> ComponentDidUpdate is called here -> State is set -> New rerender because prop change caused state change (2)

![image](https://user-images.githubusercontent.com/7877010/56986048-a4e31280-6b3e-11e9-864c-9ef9eb73ab96.png)

One option to avoid this double render is `getDerivedStateFromProps` which will update state from the props change but make sure to read up on the docs if that is truly the route to take

https://reactjs.org/docs/react-component.html?utm_source=caibaojian.com#static-getderivedstatefromprops

Reply:
I'm not a huge fan of `getDerivedStateFromProps`, but this seems to be a reasonable use case for it. The loading state is mostly controlled by render callbacks, but a change in props may also cause it to reset.

Reply:
I can do it, but maybe there's another solution. Since the child component `Image` handles the error and also calls the parent method when it occurs, maybe the responsibility to handle the reset on `Avatar` state should be on `Image`, because it already "knows" when there's an error or not.
Implementing it this way, the state on Avatar will only be changed when there's no error on `Image` and not when the src changes.
What do you think?

Reply:
I've noticed that there is an `onLoad` on `Image`, would it be ok to control the `Image` and the `DefaultAvatar` components `opacity` according to the variable?
If so, the component wouldn't need anything but to listen to both `onLoad` and `onError`.

Reply:
Did it using `getDerivedStateFromProps`, hope it fits the need 👍